tests/stdlib/tsystem.nim

```diff
@@ -74,3 +74,29 @@ template main =
 
 static: main()
 main()
+
+# bug #19967
+block:
+  type
+    X = object
+      a: string
+      b: set[char]
+
+  var y = X(b: {'a'})
+
+  reset(y)
+
+  doAssert y.b == {}
+
+block:
+  type
+    Color = enum
+      Red, Blue, Yellow
+    X = object
+      a: string
+      b: set[Color]
+
+  var fuck = X(b: {Red, Blue})
+
+  doAssert fuck.b == {}
```

Comment:
Why would it be empty? What am I missing here?

Reply:
Sorry, my mistake.

Reply:
And CI catched my mistake.